### PR TITLE
BUILD-2148 Fetch secrets from Vault

### PR DESCRIPTION
**Pull request by elrzn**

### CODE REVIEW

**Comment by jacek-poreda-sonarsource** [comment]
@elrzn Hey, you can cherry-pick this commit https://github.com/SonarSource/sonar-scanner-maven/pull/159/commits/464221333da32fa9fbca52b8a10ac284ee99c877 to fix ITs which are failing

**Review by tomverin — APPROVED** [review]

**Comment by alain-kermis-sonarsource** [comment]
Hi there guys. I see that ITs are failing. This is because of the changes we introduced in SQ [MMF-2745](https://sonarsource.atlassian.net/browse/MMF-2745), which disallows anonymous scans from now on. We introduced a method on the Orchestrator to allow us to use default admin credentials, like so:

```
ORCHESTRATOR = Orchestrator.builderEnv()
        .setSonarVersion(getSonarVersion())
        .useDefaultAdminCredentialsForBuilds(true)       <---- the orchestrator generates an admin token in the background to execute the scan
```

To utilize, please update the orchestrator in `.pom.xml` to version **3.40.0.183**.

We apologize for the inconvenience caused.

**Comment by alain-kermis-sonarsource** [comment]
Hi there guys (especially @elrzn). I see that ITs are failing. This is because of the changes we introduced in SQ [MMF-2745](https://sonarsource.atlassian.net/browse/MMF-2745), which disallows anonymous scans from now on. We introduced a method on the Orchestrator to allow us to use default admin credentials, like so:

```
ORCHESTRATOR = Orchestrator.builderEnv()
        .setSonarVersion(getSonarVersion())
        .useDefaultAdminCredentialsForBuilds(true)       <---- the orchestrator generates an admin token in the background to execute the scan
```

To utilize, the orchestrator in `.pom.xml` must be upgraded to version **3.40.0.183**.

We actually already have an [open PR](https://github.com/SonarSource/sonar-scanner-maven/pull/161) for this, so I think it is safe to merge, and then we will rebase after merge.

We apologize for the inconvenience caused.

**Review by alain-kermis-sonarsource — APPROVED** [review]
Orchestrator changes look good to me :+1:

File: .cirrus.yml [hunk 1]
```diff
@@ -6,22 +6,22 @@ gcp_credentials: ENCRYPTED[0f521f37b53602cd374f0da202d56b0160a00e7fc7f11a7d23972
...
+  ARTIFACTORY_DEPLOY_PASSWORD: VAULT[development/artifactory/token/SonarSource-sonar-scanner-maven-qa-deployer access_token]
   ARTIFACTORY_DEPLOY_REPO: sonarsource-public-qa
 
-  GCF_ACCESS_TOKEN: ENCRYPTED[!1fb91961a5c01e06e38834e55755231d649dc62eca354593105af9f9d643d701ae4539ab6a8021278b8d9348ae2ce8be!]
-  PROMOTE_URL: ENCRYPTED[!e22ed2e34a8f7a1aea5cff653585429bbd3d5151e7201022140218f9c5d620069ec2388f14f83971e3fd726215bc0f5e!]
+  GCF_ACCESS_TOKEN: VAULT[development/kv/data/promote data.token]
+  PROMOTE_URL: VAULT[development/kv/data/promote data.url]
 
-  GITHUB_TOKEN: ENCRYPTED[!f458126aa9ed2ac526f220c5acb51dd9cc255726b34761a56fc78d4294c11089502a882888cef0ca7dd4085e72e611a5!]
+  GITHUB_TOKEN: VAULT[development/github/token/SonarSource-sonar-scanner-maven-ro token]
```
[the remark this answers is not in the document]
Comment (elrzn) [reply .cirrus.yml:20]: Good point. I have already removed it in other repositories, but forgot to give it a try here 👍

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -137,13 +62,13 @@ jobs:
...
+          OSSRH_PASSWORD: ${{ fromJSON(steps.secrets.outputs.vault).ossrh_password }}
       - name: Notify on failure
         if: failure()
         uses: 8398a7/action-slack@v3
         with:
           status: failure
           fields: repo,author,eventName
         env:
-          SLACK_WEBHOOK_URL: ${{ secrets.SLACK_BUILD_WEBHOOK }}
+          SLACK_WEBHOOK_URL: ${{ fromJSON(steps.secrets.outputs.vault).slack_webhook }}
```
[the remark this answers is not in the document]
Comment (elrzn) [reply .github/workflows/release.yml:74]: Why? Shouldn't the slack_webhook contain the channel info?

**Comment by elrzn** [comment]
@malena-ebert-sonarsource Thank you for your suggestions. I've implemented them all.

**Pull request merged**